xyz-psql-connector/src/main/java/com/here/xyz/psql/DatabaseHandler.java

```diff
@@ -165,50 +171,51 @@ void reset() {
-                         appName = String.format("%s[%s]", config.applicationName(), sName );
+        if(resourceInitFinished){
+            /** Check if db-params has changed*/
+            if(!dbInstanceMap.get(dbStringConfig).getDatabaseConfig().equalsIgnoreCase(config.getConfigValuesAsString())) {
+                resourceInitFinished = false;
+                dbInstanceMap.remove(dbStringConfig);
+            }
+        }
 
-            final ComboPooledDataSource source = getComboPooledDataSource(config.host(), config.port(), config.database(), config.user(),
-                    config.password(), appName, config.maxPostgreSQLConnections());
+        if (!resourceInitFinished) {
+            /** Init dataSource, readDataSource ..*/
+            logger.info("{} - Create new config and data source for following dbStringConfig: '{}'", streamId, dbStringConfig);
+            final ComboPooledDataSource source = getComboPooledDataSource(config.getDatabaseSettings(), config.getConnectorParams(), config.applicationName() , false);
 
             Map<String, String> m = new HashMap<>();
-            m.put(C3P0EXT_CONFIG_SCHEMA, config.schema());
+            m.put(C3P0EXT_CONFIG_SCHEMA, config.getDatabaseSettings().getSchema());
             source.setExtensions(m);
 
-            final DatabaseMaintainer dbMaintainer = new DatabaseMaintainer(source,config);
-            final XYZDBInstance xyzDBInstance = new XYZDBInstance(source ,dbMaintainer, config);
+            final DatabaseMaintainer dbMaintainer = new DatabaseMaintainer(source, config);
+            final XYZDBInstance xyzDBInstance = new XYZDBInstance(source ,dbMaintainer);
 
-            if (config.replica() != null) {
-                final ComboPooledDataSource replicaDataSource = getComboPooledDataSource(config.replica(), config.port(), config.database(),
-                        config.user(), config.password(), appName, config.maxPostgreSQLConnections());
+            if (config.getDatabaseSettings().getReplicaHost() != null) {
+                final ComboPooledDataSource replicaDataSource = getComboPooledDataSource(config.getDatabaseSettings(), config.getConnectorParams(),  config.applicationName() , true);
                 replicaDataSource.setExtensions(m);
                 xyzDBInstance.addReadDataSource(replicaDataSource);
             }
-            dbInstanceMap.put(ecps, xyzDBInstance);
+            dbInstanceMap.put(dbStringConfig, xyzDBInstance);
         }
 
-        this.dataSource = dbInstanceMap.get(ecps).getDataSource();
-        this.readDataSource = dbInstanceMap.get(ecps).getReadDataSource();
-        this.dbMaintainer = dbInstanceMap.get(ecps).getDatabaseMaintainer();
-        this.config = dbInstanceMap.get(ecps).getConfig();
+        this.retryAttempted = false;
+        this.dataSource = dbInstanceMap.get(dbStringConfig).getDataSource();
```

Comment:
If reset's and initialize's bodies are not synchronized by the same object, this can lead to NPE